api/src/test/java/org/pmiops/workbench/db/dao/NewUserSatisfactionSurveyResponseDaoTest.java

```diff
@@ -0,0 +1,50 @@
+package org.pmiops.workbench.db.dao;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import javax.persistence.EntityManager;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.pmiops.workbench.FakeClockConfiguration;
+import org.pmiops.workbench.config.CommonConfig;
+import org.pmiops.workbench.db.model.DbNewUserSatisfactionSurveyResponse;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.utils.TestMockFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.context.annotation.Import;
+
+@Import({FakeClockConfiguration.class, CommonConfig.class})
+@DataJpaTest
+public class NewUserSatisfactionSurveyResponseDaoTest {
+  @Autowired private UserDao userDao;
+  @Autowired NewUserSatisfactionSurveyResponseDao newUserSatisfactionSurveyResponseDao;
+
+  private DbUser user;
+  private DbNewUserSatisfactionSurveyResponse newUserSatisfactionSurveyResponse;
+
+  @Autowired private EntityManager entityManager;
+
+  @BeforeEach
+  public void setUp() {
+    user = userDao.save(new DbUser());
+
+    newUserSatisfactionSurveyResponse =
+        TestMockFactory.createDefaultNewUserSatisfactionSurveyResponse(user);
```

Comment:
> I don't think `TestMockFactory.createDefaultNewUserSatisfactionSurveyResponse` is the right way to insert record.

I removed the method. Agreed that this shouldn't be used in the test setup since these tests may depend on no values being present in the DB.

If not here, then when or how should I use `TestMockFactory`? Other methods in that class return DB entities. In this case, the `testGetByUser` test can use a `DbNewUserSatisfactionSurveyResponse` with any properties, so I thought a default DB object was fine.

> Insert/Get/Update/Delete entity

I added a CRUD test, which acts as a nice smoke test that the `CrudRepository` is configured correctly 👍